### PR TITLE
Add setup.py, pyproject.toml

### PR DESCRIPTION
**Pull request by clbarnes**
Minimal build files to allow people to play with it locally.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from setuptools import setup
+
+requirements = [
+    "fsspec",
+    "numpy",
+    "numcodecs",
```
Comment (jakirkham) [inline setup.py:6]: Do we know what versions are needed Alistair?

Comment (alimanfoo) [reply setup.py:6]: Not sure what minimum versions would be. Do we need to pin a minimum version or can we leave this open?

**Comment by alimanfoo** [comment]
Just to say many thanks for this @clbarnes, much appreciated.

File: setup.py (hunk 1 above)
Comment (clbarnes) [reply setup.py:6]: I think they can be left open. I was in two minds about whether this should go in at all as *lack* of build files might be a useful tool to dissuade people from using it for the time being. But certainly, there should be no guarantees for support and it shouldn't be included with other projects, so loose dependencies (just enough that a clean, modern python environment could `pip install` it from github or the directory) should be fine.

**Comment by clbarnes** [comment]
I can also un-typo the commit title on request :grimacing:

File: setup.py (hunk 1 above)
Comment (jakirkham) [reply setup.py:6]: I think they can be left open. Just wanted to make sure we didn't get too old of a version by accident. Guess we could just pick latest whenever this gets tagged.

**Pull request merged**